### PR TITLE
Replaced useDipatch with useQuery/request in FacilityCreate and HospitalList files

### LINKED ISSUE #6391: 🛠️ Replace `useDispatch` w. `useQuery`/`request`: Facility (Part 2, E-H) (`src/Components/Facility/[E-H]*.tsx`)

### PR DESCRIPTION
**Pull request by sriharsh05**
### <samp>🤖 Generated by Copilot at 1aefc25</samp>

Refactored some components in the `Facility` module to use the `request` function and the `useQuery` hook for making and fetching API calls. Added and used some types to ensure type safety and consistency for the data from the backend. Removed some unnecessary code and fixed some minor issues.

The pull request partially refactors the codebase for managing facility module in the frontend. It replaces the redux actions and states with the custom request function and the useQuery hook for data fetching and updating. It improves the code quality, performance, and readability by removing unnecessary imports, variables, and dependencies.

This pull request makes the required changes to only 2 files and does not close the entire issue.

## Proposed Changes

- Fixes #6391 
- Replaced useDispatch with useQuery and request in HospitalList
- Replaced useDispatch with useQuery and request in FacilityCreate

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### <samp>🤖 Generated by Copilot at 1aefc25</samp>

*  Refactored the `FacilityCreate` component to use the `request` function and the `useQuery` hook for fetching data from the API and removed the `useAbortableEffect` hook and the `dispatchAction` function ([link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL4-R4), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL16-R18), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL40), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL60-R52), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL153), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL180-R198), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL248-R294), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL308-R319), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL523-R525), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL606-R602), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL643-R639), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL689-R685), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL724-R720))
*  Updated the `data` variable in the `FacilityCreate` component to match the API schema and removed the unnecessary properties ([link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL482-R471), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL488-R483))
*  Added a type check for the `district` property in the `fetchedDistricts` array in the `FacilityCreate` component to avoid errors ([link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL365-R359))
*  Converted the `id` variable to a string before setting it as the `createdFacilityId` state in the `FacilityCreate` component ([link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL534-R531))
*  Removed the error handling logic from the `FacilityCreate` component as it was handled by the `request` function internally ([link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-c1bd2754f233548fd1a6217ebe45fe603c71864f10c26f6350596f85b39e000cL542-L546))
*  Refactored the `HospitalList` component to use the `useQuery` hook for fetching data from the API and removed the `useAbortableEffect` hook, the `dispatchAction` function and the `totalCount` variable ([link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-8e39f154ccf13c9dedd0f968d21a0abbde450e88df9724a4e7d4264d487a7846L6-R7), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-8e39f154ccf13c9dedd0f968d21a0abbde450e88df9724a4e7d4264d487a7846R22-R23), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-8e39f154ccf13c9dedd0f968d21a0abbde450e88df9724a4e7d4264d487a7846L42-R45), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-8e39f154ccf13c9dedd0f968d21a0abbde450e88df9724a4e7d4264d487a7846L66-R79), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-8e39f154ccf13c9dedd0f968d21a0abbde450e88df9724a4e7d4264d487a7846L170-R98), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-8e39f154ccf13c9dedd0f968d21a0abbde450e88df9724a4e7d4264d487a7846L180-R109), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-8e39f154ccf13c9dedd0f968d21a0abbde450e88df9724a4e7d4264d487a7846L188-R118), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-8e39f154ccf13c9dedd0f968d21a0abbde450e88df9724a4e7d4264d487a7846L247-R182))
*  Replaced the `stateName`, `districtName` and `localbodyName` variables with the `stateData`, `districtData` and `localBodyData` variables in the `HospitalList` component to display the state, district and local body names from the data returned by the `useQuery` hook ([link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-8e39f154ccf13c9dedd0f968d21a0abbde450e88df9724a4e7d4264d487a7846L269-R200))
*  Added the `IStateListResponse` type to the `./models` file to define the type of the response returned by the API for listing the states ([link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-e2841f01e53119db63c01994d866100758f06ecc85d11914eecc7cdfe8a2f4dfR233-R237))
*  Imported the `FacilityModel`, `CapacityModal`, `DoctorModal`, `DistrictModel`, `LocalBodyModel`, `StateModel` and `IStateListResponse` types from `./models` to use in the `routes` object in the `./api` file ([link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL31-R39), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-e2841f01e53119db63c01994d866100758f06ecc85d11914eecc7cdfe8a2f4dfR6))
*  Added the `TRes` property to the `createFacility`, `updateFacility`, `getCapacity`, `listDoctor`, `statesList`, `getState`, `getDistrict`, `getDistrictByState` and `getLocalBody` routes in the `routes` object in the `./api` file to define the type of the response returned by the API for each endpoint ([link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cR238), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cR256), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cR469), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cR483), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL599-R614), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL609-R625), [link](https://github.com/coronasafe/care_fe/pull/6591/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cR644))

### CODE REVIEW

**Comment by sriharsh05** [comment]
@rithviknishad 
Please review this PR.

**Comment by rithviknishad** [comment]
@sriharsh05 the cypress is failing

**Comment by rithviknishad** [comment]
@sriharsh05

**Comment by sriharsh05** [comment]
> @sriharsh05

I'll look into it @rithviknishad

**Comment by sriharsh05** [comment]
@rithviknishad Please review the code.

File: src/Components/Facility/FacilityCreate.tsx [hunk 1]
```diff
@@ -177,36 +166,36 @@ export const FacilityCreate = (props: FacilityProps) => {
...
-        const districtList = await dispatchAction(getDistrictByState({ id }));
-        if (districtList) {
-          setDistricts([...districtList.data]);
-        }
-        setIsDistrictLoading(false);
-        return districtList ? [...districtList.data] : [];
+  const fetchDistricts = async (id: number) => {
+    if (id > 0) {
+      setIsDistrictLoading(true);
+      const { res, data } = await request(routes.getDistrictByState, {
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]: use useQuery instead

File: src/Components/Facility/FacilityCreate.tsx [hunk 2]
```diff
@@ -177,36 +166,36 @@ export const FacilityCreate = (props: FacilityProps) => {
...
-          getLocalbodyByDistrict({ id })
-        );
-        setIsLocalbodyLoading(false);
-        if (localBodyList) {
-          setLocalBodies([...localBodyList.data]);
-        }
+  const fetchLocalBody = async (id: number) => {
+    if (id > 0) {
+      setIsLocalbodyLoading(true);
+      const { res, data } = await request(routes.getLocalbodyByDistrict, {
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]: use useQuery instead

File: src/Components/Facility/FacilityCreate.tsx [hunk 3]
```diff
@@ -245,90 +234,89 @@ export const FacilityCreate = (props: FacilityProps) => {
...
-        setIsWardLoading(true);
-        const wardList = await dispatchAction(getWardByLocalBody({ id }));
-        setIsWardLoading(false);
-        if (wardList) {
-          setWard([...wardList.data.results]);
-        }
+  const fetchWards = async (id: number) => {
+    if (id > 0) {
+      setIsWardLoading(true);
+      const { res, data } = await request(routes.getWardByLocalBody, {
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]: use useQuery instead

File: src/Components/Facility/FacilityCreate.tsx [hunk 4]
```diff
@@ -245,90 +234,89 @@ export const FacilityCreate = (props: FacilityProps) => {
...
-      fetchStates(status);
-    },
-    [dispatch, fetchData]
-  );
+  useEffect(() => {
+    if (facilityId) {
+      facilityFetch();
+    }
+    fetchStates();
+  }, [dispatch]);
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]: You don't need this if you use useQuery with prefetch specified

```suggestion
```

File: src/Components/Facility/FacilityCreate.tsx [hunk 5]
```diff
@@ -245,90 +234,89 @@ export const FacilityCreate = (props: FacilityProps) => {
...
-  const fetchData = useCallback(
-    async (status: statusType) => {
+  const { refetch: facilityFetch } = useQuery(routes.getPermittedFacility, {
+    pathParams: {
+      id: facilityId || "",
+    },
+    prefetch: facilityId !== undefined,
+    onResponse: ({ res, data }) => {
       if (facilityId) {
         setIsLoading(true);
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx:237]: use the loading from the useQuery return value instead

File: src/Components/Facility/FacilityCreate.tsx [hunk 6]
```diff
@@ -245,90 +234,89 @@ export const FacilityCreate = (props: FacilityProps) => {
...
           };
           dispatch({ type: "set_form", form: formData });
           Promise.all([
-            fetchDistricts(res.data.state),
-            fetchLocalBody(res.data.district),
-            fetchWards(res.data.local_body),
+            fetchDistricts(data.state || 0),
+            fetchLocalBody(data.district || 0),
+            fetchWards(data.local_body || 0),
           ]);
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]: useQuery can handle this automatically if u specify the prefetch.

File: src/Components/Facility/FacilityCreate.tsx [hunk 7]
```diff
@@ -245,90 +234,89 @@ export const FacilityCreate = (props: FacilityProps) => {
...
     },
-    [dispatchAction, facilityId, fetchDistricts, fetchLocalBody, fetchWards]
-  );
+  });
 
-  const fetchStates = useCallback(
-    async (status: statusType) => {
+  const { refetch: fetchStates } = useQuery(routes.statesList, {
+    onResponse: ({ res, data }) => {
       setIsStateLoading(true);
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]: use loading state from the useQuery's return

File: src/Components/Facility/FacilityCreate.tsx [hunk 8]
```diff
@@ -177,36 +166,36 @@ export const FacilityCreate = (props: FacilityProps) => {
...
+          id: String(id),
+        },
+      });
+      if (res?.ok && data) {
+        setDistricts(data);
       }
-    },
-    [dispatchAction]
-  );
+      setIsDistrictLoading(false);
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]: use loading state from the useQuery's return

File: src/Components/Facility/FacilityCreate.tsx [hunk 9]
```diff
@@ -177,36 +166,36 @@ export const FacilityCreate = (props: FacilityProps) => {
...
-        }
+  const fetchLocalBody = async (id: number) => {
+    if (id > 0) {
+      setIsLocalbodyLoading(true);
+      const { res, data } = await request(routes.getLocalbodyByDistrict, {
+        pathParams: {
+          id: String(id),
+        },
+      });
+      setIsLocalbodyLoading(false);
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]: use loading state from the useQuery's return

File: src/Components/Facility/FacilityCreate.tsx [hunk 10]
```diff
@@ -245,90 +234,89 @@ export const FacilityCreate = (props: FacilityProps) => {
...
-        }
+  const fetchWards = async (id: number) => {
+    if (id > 0) {
+      setIsWardLoading(true);
+      const { res, data } = await request(routes.getWardByLocalBody, {
+        pathParams: {
+          id: String(id),
+        },
+      });
+      setIsWardLoading(false);
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]: use loading state from the useQuery's return

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by rithviknishad** [comment]
@sriharsh05 cypress tests are still failing

**Comment by sriharsh05** [comment]
@rithviknishad Please review the changes.

File: src/Components/Facility/FacilityCreate.tsx [hunk 11]
```diff
@@ -150,62 +131,45 @@ const facilityCreateReducer = (state = initialState, action: FormAction) => {
...
   const [currentStep, setCurrentStep] = useState(1);
   const [createdFacilityId, setCreatedFacilityId] = useState("");
   const [showAutoFilledPincode, setShowAutoFilledPincode] = useState(false);
   const [capacityData, setCapacityData] = useState<Array<CapacityModal>>([]);
   const [doctorData, setDoctorData] = useState<Array<DoctorModal>>([]);
   const [bedCapacityKey, setBedCapacityKey] = useState(0);
   const [docCapacityKey, setDocCapacityKey] = useState(0);
+  const [stateId, setStateId] = useState(0);
+  const [districtId, setDistrictId] = useState(0);
+  const [localBodyId, setLocalBodyId] = useState(0);
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]: `undefined` or `null` would be better to represent a not selected value.

```suggestion
  const [stateId, setStateId] = useState<number>();
  const [districtId, setDistrictId] = useState<number>();
  const [localBodyId, setLocalBodyId] = useState<number>();
```

File: src/Components/Facility/FacilityCreate.tsx [hunk 12]
```diff
@@ -150,62 +131,45 @@ const facilityCreateReducer = (state = initialState, action: FormAction) => {
...
-      }
-    },
-    [dispatchAction]
+  const { data: districtData, loading: isDistrictLoading } = useQuery(
+    routes.getDistrictByState,
+    {
+      pathParams: {
+        id: String(stateId),
+      },
+      prefetch: stateId !== 0,
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]:
```suggestion
      prefetch: !!stateId,
```

File: src/Components/Facility/FacilityCreate.tsx [hunk 13]
```diff
@@ -150,62 +131,45 @@ const facilityCreateReducer = (state = initialState, action: FormAction) => {
...
-      }
-    },
-    [dispatchAction]
+  const { data: localbodyData, loading: isLocalbodyLoading } = useQuery(
+    routes.getLocalbodyByDistrict,
+    {
+      pathParams: {
+        id: String(districtId),
+      },
+      prefetch: districtId !== 0,
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]:
```suggestion
      prefetch: !!districtId,
```

File: src/Components/Facility/FacilityCreate.tsx [hunk 14]
```diff
@@ -245,89 +209,68 @@ export const FacilityCreate = (props: FacilityProps) => {
...
-      }
-    },
-    [dispatchAction]
+  const { data: wardData, loading: isWardLoading } = useQuery(
+    routes.getWardByLocalBody,
+    {
+      pathParams: {
+        id: String(localBodyId),
+      },
+      prefetch: localBodyId !== 0,
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]:
```suggestion
      prefetch: !!localBodyId,
```

File: src/Components/Facility/FacilityCreate.tsx [hunk 15]
```diff
@@ -245,89 +209,68 @@ export const FacilityCreate = (props: FacilityProps) => {
...
+    }
   );
 
-  const fetchData = useCallback(
-    async (status: statusType) => {
+  useQuery(routes.getPermittedFacility, {
+    pathParams: {
+      id: facilityId || "",
+    },
+    prefetch: facilityId !== undefined,
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]:
```suggestion
    prefetch: !!facilityId,
```

File: src/Components/Facility/FacilityCreate.tsx [hunk 16]
```diff
@@ -245,89 +209,68 @@ export const FacilityCreate = (props: FacilityProps) => {
...
+      },
+      prefetch: localBodyId !== 0,
+    }
   );
 
-  const fetchData = useCallback(
-    async (status: statusType) => {
+  useQuery(routes.getPermittedFacility, {
+    pathParams: {
+      id: facilityId || "",
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]:
```suggestion
      id: facilityId!,
```

File: src/Components/Facility/FacilityCreate.tsx [hunk 17]
```diff
@@ -245,89 +209,68 @@ export const FacilityCreate = (props: FacilityProps) => {
...
+            pincode: data.pincode ? data.pincode : "",
+            phone_number: data.phone_number
+              ? data.phone_number.length == 10
+                ? "+91" + data.phone_number
+                : data.phone_number
+              : "",
+            latitude: data.location ? String(data.location.latitude) : "",
+            longitude: data.location
+              ? String(data.location.latitude)
+              : "" || "",
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]: 1. Are you sure this is a nested JSON? Have you tested it?
2. Why is longitude set to latitude?

```suggestion
            latitude: data.latitude,
            longitude: data.longitude,
```

File: src/Components/Facility/FacilityCreate.tsx [hunk 18]
```diff
@@ -245,89 +209,68 @@ export const FacilityCreate = (props: FacilityProps) => {
...
           };
           dispatch({ type: "set_form", form: formData });
-          Promise.all([
-            fetchDistricts(res.data.state),
-            fetchLocalBody(res.data.district),
-            fetchWards(res.data.local_body),
-          ]);
+          setStateId(data.state || 0);
+          setDistrictId(data.district || 0);
+          setLocalBodyId(data.local_body || 0);
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]: You won't need to do this coalescing if you had undefined/null as uninitialized value

```suggestion
          setStateId(data.state);
          setDistrictId(data.district);
          setLocalBodyId(data.local_body);
```

File: src/Components/Facility/FacilityCreate.tsx [hunk 19]
```diff
@@ -479,19 +424,20 @@ export const FacilityCreate = (props: FacilityProps) => {
...
         features: state.form.features,
         ward: state.form.ward,
-        kasp_empanelled: JSON.parse(state.form.kasp_empanelled),
-        latitude: state.form.latitude || null,
-        longitude: state.form.longitude || null,
+        pincode: state.form.pincode,
+        location: {
+          latitude: Number(state.form.latitude),
+          longitude: Number(state.form.longitude),
+        },
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]: Are you sure coordinates is a nested JSON and is correct? Have you tested it? Kindly test your PR before requesting review!

```suggestion
        latitude: state.form.latitude || null,
        longitude: state.form.longitude || null,
```

File: src/Redux/api.tsx [hunk 20]
```diff
@@ -634,19 +644,23 @@ const routes = {
   // States
   statesList: {
     path: "/api/v1/state/",
+    TRes: Type<IStateListResponse>(),
```
Comment (rithviknishad) [inline src/Redux/api.tsx]: Wrap it as `PaginatedResponse` inside this file! It'd be easier to read this way.

```suggestion
    TRes: Type<PaginatedResponse<StateModel>>(),
```

File: src/Redux/api.tsx [hunk 21]
```diff
@@ -634,19 +644,23 @@ const routes = {
...
 
   // Districts
 
   getDistrict: {
     path: "/api/v1/district/{id}/",
+    TRes: Type<DistrictModel>(),
   },
   getDistrictByState: {
     path: "/api/v1/state/{id}/districts/",
+    TRes: Type<StateModel[]>(),
```
Comment (rithviknishad) [inline src/Redux/api.tsx]: Why is districts API returning sates list?

```suggestion
    TRes: Type<DistrictModel[]>(),
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by sriharsh05** [comment]
@rithviknishad Please review the changes.

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
@sriharsh05 
image

- [ ] The autofill for the pincode isn't working initially in the first input field, but it works properly when you erase the initial input and enter another pincode.

image

- [ ] ward list is not being fully loaded

File: src/Utils/request/useQuery.ts [hunk 22]
```diff
@@ -31,8 +31,10 @@ export default function useQuery<TData>(
           : options;
 
       setLoading(true);
+      const response = await request(route, resolvedOptions);
       setResponse(await request(route, resolvedOptions));
```
Comment (rithviknishad) [inline src/Utils/request/useQuery.ts]:
```suggestion
      const response = await request(route, resolvedOptions);
      setResponse(response);
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]
this would end up firing the same request two times

**Comment by sriharsh05** [comment]
@rithviknishad Please review the changes.

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
@sriharsh05 fix the cypress failure

**Comment by nihal467** [comment]
@sriharsh05 clear the merge conflict

**Comment by sriharsh05** [comment]
@rithviknishad 
Please review the changes.

**Comment by sriharsh05** [comment]
@rithviknishad 
The autofill for the Pincode is not working. The API request for it is not providing the correct response. It's also not functioning properly in the deployment at https://care.ohc.network/facility/create.

**Comment by rithviknishad** [comment]
AFAIK the pincode API was down yesterday.

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
@sriharsh05 can you clear the merge conflict

**Comment by sriharsh05** [comment]
@rithviknishad 
Please review the changes.

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
image

- the above screenshot is the deploy preview

image

- the above screenshot is the current staging

- [ ] @sriharsh05 As you can see, in the deploy preview the facility homepage is not loaded properly as the total facility card loading state is missing, fix the issue

**Comment by sriharsh05** [comment]
@rithviknishad 
Please review the changes.

**Review by rithviknishad — APPROVED** [review]

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**